### PR TITLE
Adding DATA_LENGTH. Removing unused code.

### PR DESCRIPTION
**Pull request by JWhitleyAStuff**
DATA_LENGTH is a property which defines the length of the data payload
on a given CAN message. This is now used to initialize command message
data storage.

When I made the driver "DLC-agnostic," I was really only concerned with the reading of messages. This fixes the writing of messages to match the new DLCs in the DBC.

- [x] Tested in vehicle

### CODE REVIEW

**Comment by JWhitleyAStuff** [comment]
> Coming at this with almost no context, I can only make a few simple observations:
> 
> 1. Code is consistent and orderly, so if I did have some context, it will be easier to read.  Nice!
Thanks!
> 2. There are very few comments, which make reading it w/o explanation difficult. Not so nice.
On my next PR, I'll add more comments which will help explain some things.
> 3. Most deletions are either declarations, commented out code, or non safety critical systems. Okay.
I tried to get rid of the unused systems.
> 4. There is a deletion that initializes data to zeros. I'm concerned about this.
The constructor of the LockedData class now sets the length of its `data_` member (the vector where the data are actually stored) and initializes all values to 0. This is safer and keeps you from having to do it manually.
> 5. There is the deletion of "locked" data.  I'm concerned about this.
The objects that were getting created as `std::shared_ptr<LockedData>` were then just used in populating the `std::unordered_map` (`rx_list`). Now, instead, I just create the `LockedData` objects in the `insert` statements rather than creating temporary variables.
> 
> My only suggestion would be that I could probably give more input if I had more context. Maybe a merge party would help?

See comments above.

**Comment by icolwell-as** [comment]
What does "DLC" stand for?

**Comment by asaxsma** [comment]
@icolwell-as  DLC is ( Data Length Code) which is CAN lingo for the number of bytes in a CAN message, which can be 0-8.  Prior to 6.0.0 of the CAN API, the DLC for all CAN messages were set to 8 probably out of expediency.  CAN traffic on the CAN bus is on the rise, as is the the need for more message IDs increases.  One way to control traffic is to rightsize the length of messages.

**Comment by icolwell-as** [comment]
Thanks @asaxsma! that clears things up for me!

File: src/pacmod3_core.cpp [hunk 1]
```diff
@@ -29,6 +29,11 @@ const uint32_t AS::Drivers::PACMod3::TurnSignalCmdMsg::CAN_ID = 0x130;
 const uint32_t AS::Drivers::PACMod3::WiperCmdMsg::CAN_ID = 0x134;
 const uint32_t AS::Drivers::PACMod3::RearPassDoorCmdMsg::CAN_ID = 0x140;
 
+constexpr uint8_t AS::Drivers::PACMod3::SystemCmdBool::DATA_LENGTH;
```
Comment (icolwell-as) [inline src/pacmod3_core.cpp]: Why do you need to declare this again here if it was declared in pacmod3_core as `static`?

Comment (JWhitleyAStuff) [reply src/pacmod3_core.cpp]: `static` data members that are declared in a header file (regardless of whether they have a default value or not) have to have an implementation in the `cpp` file or the linker will complain. Here's a more detailed explanation: https://softwareengineering.stackexchange.com/questions/145299/why-the-static-data-members-have-to-be-defined-outside-the-class-separately-in-c

**Comment by JWhitleyAStuff** [comment]
Tested in Lexus 450h.

**Pull request merged**